kinesis_manager/include/kinesis_manager/kinesis_stream_manager.h

```diff
@@ -34,38 +34,37 @@ namespace Kinesis {
  * Constructs parameter paths for AWS Kinesis applications.
  * @param stream_idx
  * @param parameter_name
- * @return string representing the parameter path, i.e.
+ * @return ParameterPath object representing the parameter path, i.e.
  * kinesis_video/stream<stream_idx>/<parameter_name>
  * @note calling GetStreamParameterPath with negative stream index would result in construction of a
  * global parameter path, rather than one that's related to a specific stream. This is appropriate
  * for parameters such as "stream_count".
  */
-inline std::string GetStreamParameterPath(int stream_idx, const char * parameter_name)
+inline ParameterPath GetStreamParameterPath(int stream_idx, const char * parameter_name)
 {
-  std::stringstream stream_parameter_prefix;
-  stream_parameter_prefix << std::string(kStreamParameters.prefix);
+  ParameterPath path(kStreamParameters.prefix);
   if (INVALID_STREAM_ID != stream_idx) {
-    stream_parameter_prefix << std::string(kStreamParameters.stream_namespace)
-                            << std::to_string(stream_idx) << "/";
+    path += kStreamParameters.stream_namespace + std::to_string(stream_idx);
```

Comment:
Looks like you dropped a `"/"` on the end here. Do we not need this anymore between the stream_idx and the parameter_name?

Reply:
The ParameterPath object only takes in path components. It would be wrong to give it the separators.

I did have to modify some unit test (in a downstream package) to account for this change.